src/hardware/serial_terminal.rs

```diff
@@ -25,28 +75,126 @@ impl Write for OutputBuffer {
     }
 }
 
+impl core::fmt::Write for Context {
+    /// Write data to the serial terminal.
+    ///
+    /// # Note
+    /// The terminal uses an internal buffer. Overflows of the output buffer are silently ignored.
+    fn write_str(&mut self, s: &str) -> core::fmt::Result {
+        self.output.write_str(s)
+    }
+}
+
+fn handle_property_read(
+    _menu: &menu::Menu<Context>,
+    item: &menu::Item<Context>,
+    args: &[&str],
+    context: &mut Context,
+) {
+    let props: heapless::Vec<&'_ str, 2> = if let Some(prop) =
+        menu::argument_finder(item, args, "property").unwrap()
+    {
+        heapless::Vec::from_slice(&[prop]).unwrap()
+    } else {
+        heapless::Vec::from_slice(&["id", "broker"]).unwrap()
+    };
+
+    for prop in props {
```

Comment:
I'm not a huge fan of this. Maybe we just make the wrapper type a constant 255 buffer so we can maintain a standard type for both objects and then can simplify the iteration.

Reply:
Only difficulty is somehow assigning different keys to each object